Makefile

```diff
@@ -37,72 +41,98 @@ all: build
 
 clean:
 	rm -rf $(BUILD_DIR)
-	git submodule update --init --recursive
 
 # Build Dependencies (K Submodule)
 # --------------------------------
 
-deps: $(K_SUBMODULE)/make.timestamp $(TANGLER)
+K_JAR := $(K_SUBMODULE)/k-distribution/target/release/k/lib/java/kernel-1.0-SNAPSHOT.jar
 
-$(K_SUBMODULE)/make.timestamp:
-	git submodule update --init --recursive
-	cd $(K_SUBMODULE) && mvn package -DskipTests -Dproject.build.type=$(K_BUILD_TYPE)
-	touch $(K_SUBMODULE)/make.timestamp
+deps: $(K_JAR) $(TANGLER)
 
-$(TANGLER):
-	git submodule update --init -- $(PANDOC_TANGLE_SUBMODULE)
+$(K_JAR):
+	cd $(K_SUBMODULE) && mvn package -DskipTests -Dproject.build.type=$(K_BUILD_TYPE)
 
 # Building Definition
 # -------------------
 
-MAIN_MODULE        := WASM-TEST
-MAIN_SYNTAX_MODULE := WASM-TEST-SYNTAX
-MAIN_DEFN_FILE     := test
+SOURCE_FILES       := data         \
+                      kwasm-lemmas \
+                      numeric      \
+                      test         \
+                      wasm         \
+                      wasm-text    \
+                      wrc20
+EXTRA_SOURCE_FILES :=
+ALL_FILES          := $(patsubst %, %.k, $(SOURCE_FILES) $(EXTRA_SOURCE_FILES))
+
+defn:  defn-haskell defn-llvm
+build: build-llvm build-haskell
+
+KOMPILE_OPTS :=
```

Comment:
I don't like moving variables too far away from where they are used, so that you're not tempted to use them elsewhere.

But I'll move it to the top of eth Building Definitions section.